client/src/components/Layout/Navbar.tsx

```diff
@@ -0,0 +1,217 @@
+import MenuItem from '@material-ui/core/MenuItem';
+import Popover from '@material-ui/core/Popover';
+import { Link } from 'react-router-dom';
+import Toolbar from '@material-ui/core/Toolbar';
+import profileImage from '../../Images/775db5e79c5294846949f1f55059b53317f51e30.png';
+import logo from '../../Images/logo.png';
+import { makeStyles } from '@material-ui/core/styles';
+import { Typography } from '@material-ui/core';
+import { useAuth } from '../../context/useAuthContext';
+
+// Placeholders simulating data from DB
+const notifications: ActiveNotification[] = [
+  {
+    title: 'Notification 1 Title',
+    subtitle: 'Notification 1 Subtitle',
+    date: '9/9/2000',
+  },
+  {
+    title: 'Notification 2 Title',
+    subtitle: 'Notification 2 Subtitle',
+    date: '8/8/2000',
+  },
+];
+
+const useStyles = makeStyles((theme) => ({
+  navMain: {
+    backgroundColor: 'white',
+    justifyContent: 'space-between',
+    position: 'static',
+  },
+  logo: {
+    margin: '1rem',
+  },
+  linksAvatarContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  links: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  linkItem: {
+    color: 'black',
+    paddingLeft: '.5rem',
+    paddingRight: '.5rem',
+    fontSize: '1rem',
+    textDecoration: 'none',
+    fontWeight: 'bold',
+  },
+  notificationContainer: {
+    paddingBottom: '.5rem',
+    paddingTop: '.5rem',
+  },
+  notificationTitle: {
+    fontWeight: 'bold',
+    fontSize: '1rem',
+  },
+  notificationSubtitle: {
+    fontSize: '.75rem',
+  },
+  notificationDate: {
+    fontSize: '.75rem',
+    fontWeight: 'bold',
+  },
+  avatar: {
+    justifySelf: 'flex-end',
+    margin: '.5rem',
+  },
+  popover: {
+    pointerEvents: 'none',
+  },
+  paper: {
+    backgroundColor: 'red',
+  },
+}));
+
+interface ActiveNotification {
+  title: string | null;
+  subtitle: string | null;
+  date: string | null;
+}
+
+interface NotificationProps {
+  titleAnchor: HTMLElement | null;
+  activeNotifications: ActiveNotification[];
+}
+
+const NotificationPopover: React.FC<NotificationProps> = ({ titleAnchor, activeNotifications }) => {
+  const classes = useStyles();
+  return (
+    <Popover
+      PaperProps={{
+        style: {
+          borderTop: '.25rem solid black',
+          borderRadius: '0px',
+          width: '300px',
+        },
+      }}
+      open={Boolean(titleAnchor)}
+      anchorEl={titleAnchor}
+      anchorOrigin={{
+        vertical: 'bottom',
+        horizontal: 'center',
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'center',
+      }}
+      className={classes.popover}
+    >
+      <MenuList autoFocusItem={Boolean(titleAnchor)}>
+        <MenuItem>
+          <Link className={classes.linkItem} to="/notifications">
+            {activeNotifications.map((notification) =>
+              notification.title !== null ? (
+                <Grid key={null} className={classes.notificationContainer}>
```

Comment:
keys are very important for React to function properly, always try to set a unique key. worst case scenario you can try using the index. In this case either a notification id, or the index combined with the title (eg. `key={'${notification.title}-${index}'}`) would work